src/jit/morph.cpp

```diff
@@ -7060,7 +7060,13 @@ void Compiler::fgMorphTailCall(GenTreeCall* call)
         }
 #endif // _TARGET_X86_
 
+#if !defined(_TARGET_X86_)
         if (call->NeedsNullCheck())
+#else
+        // When targeting x86, we always need to perform a null check on the `this` argument before tail calling to a
+        // virtual dispatch stub.
```

Comment:
> Can you add to the comment a description of why we need to do this?

I wish I could. Neither the legacy backend nor the JIT32 backend offer any explanation.

>Also, I'd prefer you write it as:

Okay.